app/javascript/components/ActionItemForm.jsx

```diff
@@ -0,0 +1,403 @@
+      if (
+        title === '' ||
+        validator.isEmpty(title, { ignore_whitespace: true })
+      ) {
+        errorMessage = 'Title is required';
+      }
+    }
+    this.setState(prevState => ({
+      errors: { ...prevState.errors, [field]: errorMessage },
+    }));
+  };
+
+  handleChange = name => event => {
+    const { value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleInternalChange = name => event => {
+    this.setState(
+      { [name]: event.target.checked }
+    );
+  };
+
+  handleDescriptionChange = name => state => {
+    const value = JSON.stringify(convertToRaw(state.getCurrentContent()));
+    this.setState({ [name]: value });
+  };
+
+  handleSubmit() {
+    const { type } = this.state;
+
+    let hasErrors = false;
+    Object.keys(this.state.errors).forEach(field => {
+      this.checkErrors(field)();
+      hasErrors = hasErrors || this.state.errors[field] !== '';
+    });
+
+    if (!hasErrors) {
+      if (type === 'create') {
+        const body = {
+          title: this.state.title,
+          description: this.state.description,
+          internal: this.state.internal,
+          participant_id: this.state.participant_id,
+        };
+        
+        apiPost('/api/case_notes', { case_note: body })
+          .then(() => window.location.reload())
+          .catch(error => console.error(error));
+      } else {
+        
+        const newTitle = this.state.title;
+        const newDescription = this.state.tempDescription;
+        const newInternal = this.state.internal;
+
+        this.setState({
+          title: newTitle,
+          description: newDescription,
+          internal: newInternal,
+        });
+  
+      const body = {
+        title: this.state.title,
+        description: this.state.description,
+        internal: this.state.internal,
+        participant_id: this.state.participant_id
+      };
+      apiPatch(`/api/case_note/${this.state.id}/`, {case_note: body}).then(() => window.location.reload())
+      .catch(error => console.error(error));
+    }
+  }
+}
+button = () => {
+  let ret;
+  if (this.state.display == 'plus') {
+    ret = (
+      <button onClick={this.handleOpen} className="plus-button">
+        +
+      </button>
+    );
+  } else if (this.state.type === 'create') {
+    ret = (
+      <Button
+        className="primary-button"
+        variant="contained"
+        color="secondary"
+        onClick={this.handleOpen}
+      >
+        NEW ACTIONITEM +
+      </Button>
+    );
+  } else if (this.state.type === 'edit') {
+    ret = <MenuItem onClick={this.handleOpen}>Edit</MenuItem>;
+  } 
+  return ret;
+};
+
+
+
+    render(){
+      let description;
+      if(this.state.type = 'create') {
+        description = 'description';
+      }
+      else if (this.state.type == 'edit') {
+        description = 'newDescription'
+      }
+      let dialog;
+      if(this.state.type == 'create' || this.state.type == 'edit') {
```

Comment:
space after if and we should probably use === instead of ==  (almost all the time). == has type coercion while === doesn't is the difference. I can tell you more about it during work session and/or you can search it up!